c7n/resources/ec2.py

```diff
@@ -384,6 +385,37 @@ def process_instance(i):
 
         with self.executor_factory(max_workers=2) as w:
             list(w.map(process_instance, instances))
+
+@actions.register('create-snapshot')
+class Snapshots(BaseAction):
+    def process(self, resources):
+        volume_sets = {}
+        for resource in resources:
+            volume_sets[resource['InstanceId']] = []
+            for blockDevice in resource['BlockDeviceMappings']:
+                volume_sets[resource['InstanceId']].append(blockDevice['Ebs']['VolumeId'])
```

Comment:
need to check if the volume is ebs, instance volumes also show in block dev mapping